### PR TITLE
Update misc methods to where syntax

### PR DESCRIPTION
**Pull request by musm**

### CODE REVIEW

File: base/dsp.jl [hunk 1]
```diff
@@ -184,8 +184,8 @@ function conv2{T}(A::StridedMatrix{T}, B::StridedMatrix{T})
     end
     return C
 end
-conv2{T<:Integer}(A::StridedMatrix{T}, B::StridedMatrix{T}) = round.(Int,conv2(float(A), float(B)))
-conv2{T<:Integer}(u::StridedVector{T}, v::StridedVector{T}, A::StridedMatrix{T}) = round.(Int,conv2(float(u), float(v), float(A)))
+conv2(A::StridedMatrix{T}, B::StridedMatrix{T}) where {T<:Integer} = round.(Int,conv2(float(A), float(B)))
+conv2(u::StridedVector{T}, v::StridedVector{T}, A::StridedMatrix{T}) where {T<:Integer} = round.(Int,conv2(float(u), float(v), float(A)))
```
Comment (tkelman) [inline base/dsp.jl]: line's getting a bit long

**Comment by musm** [comment]
i686 timedout

File: base/multimedia.jl [hunk 2]
```diff
@@ -32,7 +32,7 @@ Returns a boolean value indicating whether or not the object `x` can be written
 `mime` type. (By default, this is determined automatically by the existence of the
 corresponding [`show`](@ref) method for `typeof(x)`.)
 """
-mimewritable{mime}(::MIME{mime}, x) =
+mimewritable(::MIME{mime}, x) where {mime} =
   method_exists(show, Tuple{IO, MIME{mime}, typeof(x)})
```
Comment (tkelman) [inline base/multimedia.jl]: this line should be 4-space indented

File: base/dates/parse.jl [hunk 3]
```diff
@@ -125,9 +125,9 @@ Returns a 2-element tuple `(values, pos)`:
   the passed in type.
 * `pos::Int`: The character index at which parsing stopped.
 """
-@generated function tryparsenext_internal{T<:TimeType}(
+@generated function tryparsenext_internal(
     ::Type{T}, str::AbstractString, pos::Int, len::Int, df::DateFormat, raise::Bool=false,
-)
+) where T<:TimeType
```
Comment (musm) [inline base/dates/parse.jl]: @tkelman  the line breaking here is pretty unconventional from the rest of base.  Any suggestions.

Comment (Sacha0) [inline base/dates/parse.jl]: Not `@tkelman`, but agreed re. the linebreaking being odd. Perhaps collapse the signature as you deem consistent with the rest of the codebase, e.g. to two lines with appropriate indentation of the second line?

File: base/dates/parse.jl [hunk 4]
```diff
@@ -267,17 +267,13 @@ function Base.parse(::Type{DateTime}, s::AbstractString, df::typeof(ISODateTimeF
     throw(ArgumentError("Invalid DateTime string"))
 end
 
-function Base.parse{T<:TimeType}(
-    ::Type{T}, str::AbstractString, df::DateFormat=default_format(T),
-)
+function Base.parse(::Type{T}, str::AbstractString, df::DateFormat=default_format(T),) where T<:TimeType
```
Comment (Sacha0) [inline base/dates/parse.jl]: Should that trailing comma be there?

File: base/dates/parse.jl [hunk 5]
```diff
@@ -267,17 +267,13 @@ function Base.parse(::Type{DateTime}, s::AbstractString, df::typeof(ISODateTimeF
...
     pos, len = start(str), endof(str)
     values, pos = tryparsenext_internal(T, str, pos, len, df, true)
     T(unsafe_get(values)...)
 end
 
-function Base.tryparse{T<:TimeType}(
-    ::Type{T}, str::AbstractString, df::DateFormat=default_format(T),
-)
+function Base.tryparse(::Type{T}, str::AbstractString, df::DateFormat=default_format(T),) where T<:TimeType
```
Comment (Sacha0) [inline base/dates/parse.jl]: Likewise here, should that trailing comma exist?

File: base/dates/periods.jl [hunk 6]
```diff
@@ -426,8 +426,8 @@ for i = 1:length(fixedperiod_conversions)
     end
 end
 # have to declare thusly so that diagonal dispatch above takes precedence:
-(==){T<:FixedPeriod, S<:FixedPeriod}(x::T, y::S) = (==)(promote(x, y)...)
-Base.isless{T<:FixedPeriod, S<:FixedPeriod}(x::T, y::S) = isless(promote(x, y)...)
+(==)(x::T, y::S) where {T<:FixedPeriod, S<:FixedPeriod} = (==)(promote(x, y)...)
+Base.isless(x::T, y::S) where {T<:FixedPeriod, S<:FixedPeriod} = isless(promote(x, y)...)
```
Comment (Sacha0) [inline base/dates/periods.jl]: Extra spaces in these signatures?

File: base/dates/periods.jl [hunk 7]
```diff
@@ -439,8 +439,8 @@ let vmax = typemax(Int64) ÷ 12, vmin = typemin(Int64) ÷ 12
 end
 Base.convert(::Type{Year}, x::Month) = Year(divexact(value(x), 12))
 Base.promote_rule(::Type{Year}, ::Type{Month}) = Month
-(==){T<:OtherPeriod, S<:OtherPeriod}(x::T, y::S) = (==)(promote(x, y)...)
-Base.isless{T<:OtherPeriod, S<:OtherPeriod}(x::T, y::S) = isless(promote(x, y)...)
+(==)(x::T, y::S) where {T<:OtherPeriod, S<:OtherPeriod} = (==)(promote(x, y)...)
+Base.isless(x::T, y::S) where {T<:OtherPeriod, S<:OtherPeriod} = isless(promote(x, y)...)
```
Comment (Sacha0) [inline base/dates/periods.jl]: Extra spaces here as well?

File: base/dsp.jl [hunk 8]
```diff
@@ -184,8 +184,10 @@ function conv2{T}(A::StridedMatrix{T}, B::StridedMatrix{T})
     end
     return C
 end
-conv2{T<:Integer}(A::StridedMatrix{T}, B::StridedMatrix{T}) = round.(Int,conv2(float(A), float(B)))
-conv2{T<:Integer}(u::StridedVector{T}, v::StridedVector{T}, A::StridedMatrix{T}) = round.(Int,conv2(float(u), float(v), float(A)))
+conv2(A::StridedMatrix{T}, B::StridedMatrix{T}) where {T<:Integer} =
+    round.(Int,conv2(float(A), float(B)))
```
Comment (Sacha0) [inline base/dsp.jl]: Perhaps inject the missing space following `Int,` while you're at it? :)

File: base/dsp.jl [hunk 9]
```diff
@@ -184,8 +184,10 @@ function conv2{T}(A::StridedMatrix{T}, B::StridedMatrix{T})
...
+conv2(u::StridedVector{T}, v::StridedVector{T}, A::StridedMatrix{T}) where {T<:Integer} =
+    round.(Int,conv2(float(u), float(v), float(A)))
```
Comment (Sacha0) [inline base/dsp.jl]: Likewise here, perhaps inject the missing space following `Int,`?

File: base/linalg/cholesky.jl [hunk 10]
```diff
@@ -443,7 +443,7 @@ end
 show(io::IO, C::Cholesky{<:Any,<:AbstractMatrix}) =
     (println(io, "$(typeof(C)) with factor:");show(io,C[:UL]))
 
-A_ldiv_B!{T<:BlasFloat}(C::Cholesky{T,<:AbstractMatrix}, B::StridedVecOrMat{T}) =
+A_ldiv_B!(C::Cholesky{T,<:AbstractMatrix}, B::StridedVecOrMat{T}) where {T<:BlasFloat}=
```
Comment (Sacha0) [inline base/linalg/cholesky.jl]: Missing space following the `}`?

File: base/linalg/triangular.jl [hunk 11]
```diff
@@ -482,20 +482,20 @@ for (t, uploc, isunitc) in ((:LowerTriangular, 'L', 'N'),
...
     S = typeof((zero(T)*one(T) + zero(T))/one(T))
     UpperTriangular(A_ldiv_B!(convert(AbstractArray{S}, A), eye(S, size(A, 1))))
 end
-inv{T}(A::UnitUpperTriangular{T}) = UnitUpperTriangular(A_ldiv_B!(A, eye(T, size(A, 1))))
-inv{T}(A::UnitLowerTriangular{T}) = UnitLowerTriangular(A_ldiv_B!(A, eye(T, size(A, 1))))
+inv(A::UnitUpperTriangular{T}) where {T} = UnitUpperTriangular(A_ldiv_B!(A, eye(T, size(A, 1))))
+inv(A::UnitLowerTriangular{T}) where {T} = UnitLowerTriangular(A_ldiv_B!(A, eye(T, size(A, 1))))
 
-errorbounds{T<:Union{BigFloat, Complex{BigFloat}}}(A::AbstractTriangular{T,<:StridedMatrix}, X::StridedVecOrMat{T}, B::StridedVecOrMat{T}) =
+errorbounds(A::AbstractTriangular{T,<:StridedMatrix}, X::StridedVecOrMat{T}, B::StridedVecOrMat{T}) where {T<:Union{BigFloat, Complex{BigFloat}}} =
```
Comment (Sacha0) [inline base/linalg/triangular.jl]: Extra space in the `Union`?

**Comment by musm** [comment]
@Sacha0 thanks for the review, I addressed them all

**Review by Sacha0 — APPROVED** [review]
Thanks @musm! (Absent objections, requests for time, or someone else beating me to it, I plan to merge tomorrow morning. Best!)

**Comment by musm** [comment]
fyi @Sacha0  I added another commit updating the syntax and using Vector in the linalg file

**Comment by tkelman** [comment]
Opening multiple separate PR's for this kind of thing would be preferable to appending and making a single PR any larger than be it has to be.

**Review by Sacha0 — APPROVED** [review]
lgtm for the most recent commit!

**Pull request merged**